
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>&lt;no title&gt; &#8212; samgen  documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <p>DAG = kf_rnaseq RNA-seq Workflow based on <a class="reference external" href="https://github.com/kids-first/kf-rnaseq-workflow">https://github.com/kids-first/kf-rnaseq-workflow</a>. ## Introduction This pipeline utilizes cutadapt to trim adapters from the raw reads, if necessary, and passes the reads to STAR for alignment. The alignment output is used by RSEM for gene expression abundance estimation. Additionally, Kallisto is used for quantification, but uses pseudoalignments to estimate the gene abundance from the raw data. Fusion calling is performed using Arriba and STAR-Fusion detection tools on the STAR alignment outputs. Filtering and prioritization of fusion calls is done by annoFuse. Metrics for the workflow are generated by RNA-SeQC. ### Cutadapt <a class="reference external" href="https://github.com/marcelm/cutadapt">Cutadapt v2.5</a> Cut adapter sequences from raw reads if needed. ### STAR <a class="reference external" href="https://doi.org/f4h523">STAR v2.6.1d</a> RNA-Seq raw data alignment. ### RSEM <a class="reference external" href="https://doi:10/cwg8n5">RSEM v1.3.1</a> Calculation of gene expression. ### Kallisto <a class="reference external" href="https://doi:10.1038/nbt.3519">Kallisto v0.43.1</a> Raw data pseudoalignment to estimate gene abundance. ### STAR-Fusion <a class="reference external" href="https://doi:10.1101/120295">STAR-Fusion v1.5.0</a> Fusion detection for <code class="docutils literal notranslate"><span class="pre">STAR</span></code> chimeric reads. ### Arriba <a class="reference external" href="https://github.com/suhrig/arriba/">Arriba v1.1.0</a> Fusion caller that uses <code class="docutils literal notranslate"><span class="pre">STAR</span></code> aligned reads and chimeric reads output. ### annoFuse <a class="reference external" href="https://github.com/d3b-center/annoFuse/releases/tag/v0.90.0">annoFuse 0.90.0</a> Filter and prioritize fusion calls. For more information, please see the following <a class="reference external" href="https://www.biorxiv.org/content/10.1101/839738v3">paper</a>. ### RNA-SeQC <a class="reference external" href="https://github.com/broadinstitute/rnaseqc">RNA-SeQC v2.3.4</a> Generate metrics such as gene and transcript counts, sense/antisene mapping, mapping rates, etc ## Usage ### Inputs common: <code class="docutils literal notranslate"><span class="pre">yaml</span> <span class="pre">inputs:</span> <span class="pre">sample_name:</span> <span class="pre">string</span> <span class="pre">r1_adapter:</span> <span class="pre">['null',</span> <span class="pre">string]</span> <span class="pre">r2_adapter:</span> <span class="pre">['null',</span> <span class="pre">string]</span> <span class="pre">STAR_outSAMattrRGline:</span> <span class="pre">string</span> <span class="pre">STARgenome:</span> <span class="pre">File</span> <span class="pre">RSEMgenome:</span> <span class="pre">File</span> <span class="pre">reference_fasta:</span> <span class="pre">File</span> <span class="pre">gtf_anno:</span> <span class="pre">File</span> <span class="pre">FusionGenome:</span> <span class="pre">File</span> <span class="pre">runThread:</span> <span class="pre">int</span> <span class="pre">RNAseQC_GTF:</span> <span class="pre">File</span> <span class="pre">kallisto_idx:</span> <span class="pre">File</span> <span class="pre">wf_strand_param:</span> <span class="pre">[&quot;default&quot;,</span> <span class="pre">&quot;rf-stranded&quot;,</span> <span class="pre">&quot;fr-stranded&quot;],</span> <span class="pre">doc:</span> <span class="pre">&quot;use</span> <span class="pre">'default'</span> <span class="pre">for</span> <span class="pre">unstranded/auto,</span> <span class="pre">'rf-stranded'</span> <span class="pre">if</span> <span class="pre">read1</span> <span class="pre">in</span> <span class="pre">the</span> <span class="pre">fastq</span> <span class="pre">read</span> <span class="pre">pairs</span> <span class="pre">is</span> <span class="pre">reverse</span> <span class="pre">complement</span> <span class="pre">to</span> <span class="pre">the</span> <span class="pre">transcript,</span> <span class="pre">'fr-stranded'</span> <span class="pre">if</span> <span class="pre">read1</span> <span class="pre">same</span> <span class="pre">sense</span> <span class="pre">as</span> <span class="pre">transcript&quot;</span> <span class="pre">input_type:</span> <span class="pre">[&quot;BAM&quot;,</span> <span class="pre">&quot;FASTQ&quot;],</span> <span class="pre">doc:</span> <span class="pre">&quot;Please</span> <span class="pre">select</span> <span class="pre">one</span> <span class="pre">option</span> <span class="pre">for</span> <span class="pre">input</span> <span class="pre">file</span> <span class="pre">type,</span> <span class="pre">BAM</span> <span class="pre">or</span> <span class="pre">FASTQ.&quot;</span></code> ### Bam input-specific: <code class="docutils literal notranslate"><span class="pre">yaml</span> <span class="pre">inputs:</span> <span class="pre">reads1:</span> <span class="pre">File</span></code> ### PE Fastq input-specific: <code class="docutils literal notranslate"><span class="pre">yaml</span> <span class="pre">inputs:</span> <span class="pre">reads1:</span> <span class="pre">File</span> <span class="pre">reads2:</span> <span class="pre">File</span></code> ### SE Fastq input-specific: <code class="docutils literal notranslate"><span class="pre">yaml</span> <span class="pre">inputs:</span> <span class="pre">reads1:</span> <span class="pre">File</span></code> ### Run: 1) Fastq or bam input are accepted: For PE fastq input, please enter the reads 1 file in <code class="docutils literal notranslate"><span class="pre">reads1</span></code> and the reads 2 file in <code class="docutils literal notranslate"><span class="pre">reads2</span></code>. For SE fastq input, enter the single ends reads file in <code class="docutils literal notranslate"><span class="pre">reads1</span></code> and leave <code class="docutils literal notranslate"><span class="pre">reads2</span></code> empty as it is optional. For bam input, please enter the reads file in <code class="docutils literal notranslate"><span class="pre">reads1</span></code> and leave <code class="docutils literal notranslate"><span class="pre">reads2</span></code> empty as it is optional. 2) <code class="docutils literal notranslate"><span class="pre">r1_adapter</span></code> and <code class="docutils literal notranslate"><span class="pre">r2_adapter</span></code> are OPTIONAL. If the input reads have already been trimmed, leave these as null and cutadapt step will simple pass on the fastq files to STAR. If they do need trimming, supply the adapters and the cutadapt step will trim, and pass trimmed fastqs along. 3) <code class="docutils literal notranslate"><span class="pre">wf_strand_param</span></code> is a workflow convenience param so that, if you input the following, the equivalent will propagate to the four tools that use that parameter: - <cite>default</cite>: ‘rsem_std’: null, ‘kallisto_std’: null, ‘rnaseqc_std’: null, ‘arriba_std’: null. This means unstranded or auto in the case of arriba. - <cite>rf-stranded</cite>: ‘rsem_std’: 0, ‘kallisto_std’: ‘rf-stranded’, ‘rnaseqc_std’: ‘rf’, ‘arriba_std’: ‘reverse’. This means if read1 in the input fastq/bam is reverse complement to the transcript that it maps to. - <cite>fr-stranded</cite>: ‘rsem_std’: 1, ‘kallisto_std’: ‘fr-stranded’, ‘rnaseqc_std’: ‘fr’, ‘arriba_std’: ‘yes’. This means if read1 in the input fastq/bam is the same sense (maps 5’ to 3’) to the transcript that it maps to. 4) Suggested <code class="docutils literal notranslate"><span class="pre">STAR_outSAMattrRGline</span></code>, with <strong>TABS SEPARATING THE TAGS</strong>, format is: <code class="docutils literal notranslate"><span class="pre">ID:sample_name</span> <span class="pre">LB:aliquot_id</span> <span class="pre">PL:platform</span> <span class="pre">SM:BSID</span></code> for example <code class="docutils literal notranslate"><span class="pre">ID:7316-242</span> <span class="pre">LB:750189</span> <span class="pre">PL:ILLUMINA</span> <span class="pre">SM:BS_W72364MN</span></code> 5) Suggested inputs are: - <cite>FusionGenome</cite>: [GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz](<a class="reference external" href="https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/__genome_libs_StarFv1.3/GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz">https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/__genome_libs_StarFv1.3/GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz</a>) - <cite>gtf_anno</cite>: gencode.v27.primary_assembly.annotation.gtf, location: <a class="reference external" href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.primary_assembly.annotation.gtf.gz">ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.primary_assembly.annotation.gtf.gz</a>, will need to unzip - <cite>RNAseQC_GTF</cite>: gencode.v27.primary_assembly.RNAseQC.gtf, built using <cite>gtf_anno</cite> and following build instructions [here](<a class="reference external" href="https://github.com/broadinstitute/rnaseqc#usage">https://github.com/broadinstitute/rnaseqc#usage</a>) - <cite>RSEMgenome</cite>: RSEM_GENCODE27.tar.gz, built using the <cite>reference_fasta</cite> and <cite>gtf_anno</cite>, following <cite>GENCODE</cite> instructions from [here](<a class="reference external" href="https://deweylab.github.io/RSEM/README.html">https://deweylab.github.io/RSEM/README.html</a>), then creating a tar ball of the results. - <cite>STARgenome</cite>: STAR_GENCODE27.tar.gz, created using the star_genomegenerate.cwl tool, using the <cite>reference_fasta</cite>, <cite>gtf_anno</cite>, and setting <cite>sjdbOverhang</cite> to 100 - <cite>reference_fasta</cite>: <a class="reference external" href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/GRCh38.primary_assembly.genome.fa.gz">GRCh38.primary_assembly.genome.fa</a>, will need to unzip - <cite>kallisto_idx</cite>: gencode.v27.kallisto.index, built from gencode 27 trascript fasta: <a class="reference external" href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.transcripts.fa.gz">ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.transcripts.fa.gz</a>, following instructions from <a class="reference external" href="https://pachterlab.github.io/kallisto/manual">here</a> ### Outputs: <code class="docutils literal notranslate"><span class="pre">yaml</span> <span class="pre">outputs:</span> <span class="pre">cutadapt_stats:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">cutadapt/cutadapt_stats</span> <span class="pre">#</span> <span class="pre">only</span> <span class="pre">if</span> <span class="pre">adapter</span> <span class="pre">supplied</span> <span class="pre">STAR_transcriptome_bam:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">star/transcriptome_bam_out</span> <span class="pre">STAR_sorted_genomic_bam:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">samtools_sort/sorted_bam</span> <span class="pre">STAR_sorted_genomic_bai:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">samtools_sort/sorted_bai</span> <span class="pre">STAR_chimeric_bam_out:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">samtools_sort/chimeric_bam_out</span> <span class="pre">STAR_chimeric_junctions:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">star_fusion/chimeric_junction_compressed</span> <span class="pre">STAR_gene_count:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">star/gene_counts</span> <span class="pre">STAR_junctions_out:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">star/junctions_out</span> <span class="pre">STAR_final_log:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">star/log_final_out</span> <span class="pre">STAR-Fusion_results:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">star_fusion/abridged_coding</span> <span class="pre">arriba_fusion_results:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">arriba_fusion/arriba_fusions</span> <span class="pre">arriba_fusion_viz:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">arriba_fusion/arriba_pdf</span> <span class="pre">RSEM_isoform:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">rsem/isoform_out</span> <span class="pre">RSEM_gene:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">rsem/gene_out</span> <span class="pre">RNASeQC_Metrics:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">rna_seqc/Metrics</span> <span class="pre">RNASeQC_counts:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">supplemental/RNASeQC_counts</span> <span class="pre">#</span> <span class="pre">contains</span> <span class="pre">gene</span> <span class="pre">tpm,</span> <span class="pre">gene</span> <span class="pre">read,</span> <span class="pre">and</span> <span class="pre">exon</span> <span class="pre">counts</span> <span class="pre">kallisto_Abundance:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">kallisto/abundance_out</span> <span class="pre">annofuse_filtered_fusions_tsv:</span> <span class="pre">'File?',</span> <span class="pre">outputSource:</span> <span class="pre">annoFuse_filter/filtered_fusions_tsv,</span> <span class="pre">doc:</span> <span class="pre">&quot;Filtered</span> <span class="pre">output</span> <span class="pre">of</span> <span class="pre">formatted</span> <span class="pre">and</span> <span class="pre">annotated</span> <span class="pre">Star</span> <span class="pre">Fusion</span> <span class="pre">and</span> <span class="pre">arriba</span> <span class="pre">results&quot;</span></code>
.. image:: <a class="reference external" href="https://github.com/kids-first/kf-rnaseq-workflow/blob/master/docs/kfdrc-rnaseq-workflow.png?raw=true">https://github.com/kids-first/kf-rnaseq-workflow/blob/master/docs/kfdrc-rnaseq-workflow.png?raw=true</a></p>
<blockquote>
<div><blockquote>
<div><dl class="field-list simple">
<dt class="field-odd">target</dt>
<dd class="field-odd"><p><a class="reference external" href="https://github.com/kids-first/kf-rnaseq-workflow/blob/master/docs/kfdrc-rnaseq-workflow.png?raw=true">https://github.com/kids-first/kf-rnaseq-workflow/blob/master/docs/kfdrc-rnaseq-workflow.png?raw=true</a></p>
</dd>
<dt class="field-even">alt</dt>
<dd class="field-even"><p>pipeline flowchart</p>
</dd>
</dl>
</div></blockquote>
<p># D3b annoFuse Workflow ## Introduction In this workflow, annoFuse performs standardization of StarFusion and arriba output files to retain information regarding fused genes, breakpoints, reading frame information as well as annotation from FusionAnnotator, output format description <a class="reference external" href="https://github.com/d3b-center/annoFuse/wiki#1-standardize-calls-from-fusion-callers-to-retain-information-regarding-fused-genesbreakpoints-reading-frame-information-as-well-as-annotation-from-fusionannotator">here</a>. Basic artifact filtering to remove fusions among gene paralogs, conjoined genes and fused genes found in normal samples is also performed by filtering fusions annotated by <a class="reference external" href="https://github.com/d3b-center/FusionAnnotator">FusionAnnotator</a> with “GTEx_Recurrent|DGD_PARALOGS|Normal|BodyMap|ConjoinG”. Each fusion call needs at least one junction reads support to be retained as true call. Additionally, if a fusion call has large number of spanning fragment reads compared to junction reads (spanning fragment minus junction read greater than ten), we remove these calls as potential false positives. An expression based filter is also applied, requiring a min FPKM value of 1 for the fusion genes in question. Please refer to <a class="reference external" href="https://github.com/d3b-center/annoFuse">annoFuse</a> R package for additional applications like putative oncogene annotations. ## Usage ### Inputs <code class="docutils literal notranslate"><span class="pre">yaml</span> <span class="pre">inputs:</span> <span class="pre">sample_name:</span> <span class="pre">'string',</span> <span class="pre">doc:</span> <span class="pre">&quot;Sample</span> <span class="pre">name</span> <span class="pre">used</span> <span class="pre">for</span> <span class="pre">file</span> <span class="pre">base</span> <span class="pre">name</span> <span class="pre">of</span> <span class="pre">all</span> <span class="pre">outputs&quot;</span> <span class="pre">FusionGenome:</span> <span class="pre">'File',</span> <span class="pre">doc:</span> <span class="pre">&quot;GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz&quot;,</span> <span class="pre">sbg:suggestedValue:</span> <span class="pre">'File',</span> <span class="pre">path:</span> <span class="pre">'5d9c8d04e4b0950cce147f94',</span> <span class="pre">name:</span> <span class="pre">'GRCh38_v27_CTAT_lib_Feb092018.plug-n-play.tar.gz'</span> <span class="pre">genome_untar_path:</span> <span class="pre">'string?',</span> <span class="pre">doc:</span> <span class="pre">&quot;This</span> <span class="pre">is</span> <span class="pre">what</span> <span class="pre">the</span> <span class="pre">path</span> <span class="pre">will</span> <span class="pre">be</span> <span class="pre">when</span> <span class="pre">genome_tar</span> <span class="pre">is</span> <span class="pre">unpackaged&quot;,</span> <span class="pre">default:</span> <span class="pre">&quot;GRCh38_v27_CTAT_lib_Feb092018/ctat_genome_lib_build_dir&quot;</span> <span class="pre">rsem_expr_file:</span> <span class="pre">'File',</span> <span class="pre">doc:</span> <span class="pre">&quot;gzipped</span> <span class="pre">rsem</span> <span class="pre">gene</span> <span class="pre">expression</span> <span class="pre">file&quot;</span> <span class="pre">arriba_output_file:</span> <span class="pre">'File',</span> <span class="pre">doc:</span> <span class="pre">&quot;Output</span> <span class="pre">from</span> <span class="pre">arriba,</span> <span class="pre">usually</span> <span class="pre">extension</span> <span class="pre">arriba.fusions.tsv&quot;</span> <span class="pre">col_num:</span> <span class="pre">'int?',</span> <span class="pre">doc:</span> <span class="pre">&quot;column</span> <span class="pre">number</span> <span class="pre">in</span> <span class="pre">file</span> <span class="pre">of</span> <span class="pre">fusion</span> <span class="pre">name.&quot;</span> <span class="pre">star_fusion_output_file:</span> <span class="pre">'File',</span> <span class="pre">doc:</span> <span class="pre">&quot;Output</span> <span class="pre">from</span> <span class="pre">arriba,</span> <span class="pre">usually</span> <span class="pre">extension</span> <span class="pre">STAR.fusion_predictions.abridged.coding_effect.tsv&quot;</span> <span class="pre">output_basename:</span> <span class="pre">'string',</span> <span class="pre">doc:</span> <span class="pre">&quot;String</span> <span class="pre">to</span> <span class="pre">use</span> <span class="pre">as</span> <span class="pre">basename</span> <span class="pre">for</span> <span class="pre">outputs&quot;</span></code> ### Run 1) Outputs from the arriba and STAR Fusion runs are required ahead of time (main RNAseq worflow output) 2) Gzipped rsem counts file, also generated in main RNAseq workflow 3) <code class="docutils literal notranslate"><span class="pre">FusionGenome</span></code> should match what was used to run STAR Fusion ### Outputs <code class="docutils literal notranslate"><span class="pre">yaml</span> <span class="pre">outputs:</span> <span class="pre">annofuse_filtered_fusions_tsv:</span> <span class="pre">File,</span> <span class="pre">outputSource:</span> <span class="pre">annoFuse_filter/filtered_fusions_tsv,</span> <span class="pre">doc:</span> <span class="pre">&quot;Filtred</span> <span class="pre">output</span> <span class="pre">of</span> <span class="pre">formatted</span> <span class="pre">and</span> <span class="pre">annotated</span> <span class="pre">Star</span> <span class="pre">Fusion</span> <span class="pre">and</span> <span class="pre">arriba</span> <span class="pre">results&quot;</span></code></p>
</div></blockquote>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">samgen</a></h1>








<h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="dags.html">DAGs</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, Agnieszka Szmurło, Tomasz Gambin, Mateusz Dawidziuk, Dominika Maziec, Marek Wiewiórka.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.1.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../_sources/dags/kf_rnaseq-desc.md.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>