
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Introduction &#8212; samgen 0.2.38-9-g335b43c documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="introduction">
<h1>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h1>
<p>Germline exome workflow based on <a class="reference external" href="https://github.com/sysamgen/analysis-workflows/blob/master/definitions/pipelines/germline_exome.cwl">mgi-germline-exome</a>.</p>
<p>This worfklow completes exome alignment and germline variant detection using a variety of tools.
The initial processing of sequence is performed by picard, alignment is performed by bwa. Output bam file undergoes qc process resulting in
insert_size_metrics, alignment_summary_metrics, hs_metrics and flagstat. VerifyBamId step checks if reads are not contaminated as a mixture of two samples.
Variants are detected using GATK HaplotypeCaller tool.</p>
</section>
<section id="alignment-and-qc-step">
<h1>Alignment and QC step<a class="headerlink" href="#alignment-and-qc-step" title="Permalink to this headline">¶</a></h1>
<p>Germline pipeline starts with either fastq files or unaligned BAM files with necessary metadata describing analysis type, sample name and phenotype.</p>
<img alt="../_images/alignment_qc.png" src="../_images/alignment_qc.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="10.1093/bioinformatics/btp324">bwa_mem 0.7.15</a> - for sequence alignment step using GRCh38 reference</p></li>
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btu314">samblaster 0.1.24</a> - for adding mate tags</p></li>
<li><p><a class="reference external" href="10.3390/biology1030895">flexbar 3.5</a> - for bar and adapter removal</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard:2.23.6</a> for alignment metrics</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btp352">samtools 1.11–h6270b1f_0</a> for BAM indexing and flagstat operation</p></li>
<li><p><a class="reference external" href="http://broadinstitute.github.io/picard/">picard 2.18.1</a> - for bedtointervals operation</p></li>
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk 4.1.8.1</a>  for Select Variants</p></li>
<li><p><a class="reference external" href="doi:10.1016/j.ajhg.2012.09.004">verifyBamId v1.1.3</a> - for checking whether the reads are contaminated as a mixture of two samples</p></li>
<li><p><a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a> - for bam QC report</p></li>
<li><p><a class="reference external" href="10.1093/bioinformatics/btv566">qualimap</a> - for bam QC report</p></li>
</ul>
</section>
<section id="detect-variants-step">
<h1>Detect variants step<a class="headerlink" href="#detect-variants-step" title="Permalink to this headline">¶</a></h1>
<img alt="../_images/detect_variants.png" src="../_images/detect_variants.png" />
<p>Used tools:</p>
<ul class="simple">
<li><p><a class="reference external" href="10.1101/gr.107524.110">gatk:4.1.8.1</a> Haplotype Caller for variant (SNPs and indels) detection</p></li>
<li><p><a class="reference external" href="10.1093/gigascience/giab008">bcftools</a> - QC for detected variants</p></li>
</ul>
</section>
<section id="variants-post-processing">
<h1>Variants post-processing<a class="headerlink" href="#variants-post-processing" title="Permalink to this headline">¶</a></h1>
<blockquote>
<div><p>After variants are detected by Haplotype Caller they are subject  decomposition and normalization to assure coherent variant representation.
Once variants are standardized they are anootated with metadata collected and publishe by public databases.</p>
<p>Used tools:</p>
</div></blockquote>
<ul class="simple">
<li><p><a class="reference external" href="https://doi.org/10.1093/bioinformatics/btv112">vt 0.57721–hf74b74d_1</a> - for normalization and decomposition</p></li>
<li><p><a class="reference external" href="doi:10.1186/s13059-016-0974-4">vep</a> - for variants annotations</p></li>
</ul>
</section>
<section id="loading-variants-to-database">
<h1>Loading variants to database<a class="headerlink" href="#loading-variants-to-database" title="Permalink to this headline">¶</a></h1>
<p>After succesfull completion of variants annotations data are transformed into tabular form and stored in relational database using custom operations.</p>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../index.html">samgen</a></h1>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview/overview.html">System overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="dags.html">Pipelines overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tech_docs/tech_docs.html">Running pipelines</a></li>
<li class="toctree-l1"><a class="reference internal" href="../catalogue/catalogue.html">Data catalogue</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2021, Agnieszka Szmurło, Tomasz Gambin, Mateusz Dawidziuk, Dominika Maziec, Marek Wiewiórka.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.1.2</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../_sources/dags/germline-exome-desc.md.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>